tdrs-backend/tdpservice/data_files/views.py

```diff
@@ -147,9 +147,89 @@ def download(self, request, pk=None):
+
+            # All cat3 errors associated with FAMILY_AFFILIATION and CITIZENSHIP_STATUS
+            filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                "FAMILY_AFFILIATION",
+                "CITIZENSHIP_STATUS"
+                ],
+                error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+            if is_active:
+                # All cat3 errors associated with summed fields: AMT_FOOD_STAMP_ASSISTANCE, AMT_SUB_CC, CASH_AMOUNT,
+                # CC_AMOUNT, TRANSP_AMOUNT
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "AMT_FOOD_STAMP_ASSISTANCE", "AMT_SUB_CC", "CASH_AMOUNT", "CC_AMOUNT", "TRANSP_AMOUNT"
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+                # All cat3 errors associated with FAMILY_AFFILIATION and SSN and CITIZENSHIP_STATUS
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "FAMILY_AFFILIATION",
+                    "SSN",
+                    "CITIZENSHIP_STATUS"
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+                # All cat3 errors associated with FAMILY_AFFILIATION and PARENT_MINOR_CHILD
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "FAMILY_AFFILIATION",
+                    "PARENT_MINOR_CHILD",
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+                # All cat3 errors associated with FAMILY_AFFILIATION and EDUCATION_LEVEL
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "FAMILY_AFFILIATION",
+                    "EDUCATION_LEVEL",
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+                # All cat3 errors associated with FAMILY_AFFILIATION and WORK_ELIGIBLE_INDICATOR
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "FAMILY_AFFILIATION",
+                    "WORK_ELIGIBLE_INDICATOR",
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
+
+                # All cat3 errors associated with CITIZENSHIP_STATUS and WORK_ELIGIBLE_INDICATOR
+                filtered_errors = filtered_errors.union(all_errors.filter(fields_json__friendly_name__has_keys=[
+                    "CITIZENSHIP_STATUS",
+                    "WORK_ELIGIBLE_INDICATOR",
+                    ],
+                    error_type=ParserErrorCategoryChoices.VALUE_CONSISTENCY))
```

Comment:
is there a way to shorten this function? perhaps a function that does `filtered_errors.union` and takes a list param for `fields_json__friendly_name__has_keys` and an `error_type` param. or different functions for different user roles, for `is_active` and `is_closed` etc. 

just a lot of lines here, figured we could reduce the cognitive complexity somehow.